src/trio/_core/_run.py

```diff
@@ -2013,11 +2013,33 @@ def task_exited(self, task: Task, outcome: Outcome[object]) -> None:
                 lot.break_lot(task)
             del GLOBAL_PARKING_LOT_BREAKER[task]
 
-        if (
+        if task._child_nurseries:
+            # Forcefully abort any tasks spawned by the misnested nursery to
+            # avoid internal errors.
+            runner = GLOBAL_RUN_CONTEXT.runner
+            for nursery in task._child_nurseries:
+                nursery.cancel_scope.cancel()
+                for child in nursery._children:
+                    if child in runner.runq:
+                        runner.runq.remove(child)
+                        runner.tasks.remove(child)
+                nursery._children.clear()
+            try:
+                # Raise this, rather than just constructing it, to get a
+                # traceback frame included
+                raise RuntimeError(
+                    "Nursery stack corrupted: nurseries spawned by "
+                    f"{task!r} was still live when the task exited\n{MISNESTING_ADVICE}",
+                )
```

Comment:
I'd be inclined to make this some kind of `BaseException`, to make it really clear that this is not a recoverable error.

Reply:
the cancel scope misnesting error is `RuntimeError`, so I think this should be the same. With cleaning up aborted tasks I think this is technically recoverable, the runtime loop is not corrupted beyond redemption or anything.

So unless we want to do a thorough replacing of a lot of `RuntimeError` I think we're better off staying consistent.